polaris/evaluate/utils.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+import pandas as pd
+from typing import Optional
+
+from polaris.evaluate import BenchmarkResults, ResultsType
+from polaris.utils.types import PredictionsType
+from polaris.evaluate import Metric
+
+
+def is_multi_task_single_test_set(vals: PredictionsType, target_cols: list[str]):
+    """Check if the given values are for a multiple-task benchmark with a single
+    test set. This is inferred by comparing the target names with the keys of the
+    given data. If all keys in the given data match the target column names, we
+    assume they are target names (as opposed to test set names for a single-task,
+    multiple test set benchmark)."""
+    return all(k in target_cols for k in vals)
+
+
+def normalize_predictions_type(vals: PredictionsType, target_cols: list[str]):
+    if isinstance(vals, dict):
+        if is_multi_task_single_test_set(vals, target_cols):
+            return {"test": vals}
+        else:
+            return vals
```

Comment:
**Question:** Should we differentiate multi-test, single-task and single-test, multi-task here? Seems we would want to standardize these two cases to a 2-level dictionary as well (i.e. `dict[str, dict[str, np.ndarray]]`)

Reply:
I think we should do this another PR. I'll create a ticket.